### PR TITLE
bump ENVTEST_K8S_VERSION

### PR DESCRIPTION
**Pull request by zirain**
follow up: https://github.com/envoyproxy/gateway/pull/4169

### CODE REVIEW

File: tools/make/kube.mk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
 ENVTEST_K8S_VERSION ?= 1.28.0
 # Need run cel validation across multiple versions of k8s
-ENVTEST_K8S_VERSIONS ?= 1.27.1 1.28.0 1.29.0
+ENVTEST_K8S_VERSIONS ?= 1.28.0 1.29.0 1.30.0 1.31.0
```
Comment (arkodg) [inline tools/make/kube.mk:4]: can we pick the latest patch versions here ?
```
1.28.13 1.29.8 1.30.4 1.31.0
```

Comment (zirain) [reply tools/make/kube.mk:4]: I tried(but failed), it's actual `kubebuilder` versions.

Comment (arkodg) [reply tools/make/kube.mk:6]: ah thanks for sharing that link, here's what I see in the envtest releases
```
1.28.3 1.29.5 1.30.3 1.31.0
```

**Review by arkodg — APPROVED** [review]
thanks !

**Pull request merged**